src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -44,18 +45,31 @@ public partial class HubConnection
         private readonly IServiceProvider _serviceProvider;
         private readonly IConnectionFactory _connectionFactory;
         private readonly ConcurrentDictionary<string, InvocationHandlerList> _handlers = new ConcurrentDictionary<string, InvocationHandlerList>(StringComparer.Ordinal);
+        private DateTime _nextActivationServerTimeout;
+        private DateTime _nextActivationSendPing;
         private bool _disposed;
 
         // Transient state to a connection
         private ConnectionState _connectionState;
 
         public event Func<Exception, Task> Closed;
 
+        // internal for testing purposes
+        internal TimeSpan TickRate { get; set; } = TimeSpan.FromSeconds(1);
+
+        /// <summary>
+        /// Gets or sets the server timeout interval for the connection. 
+        /// </summary>
+        public TimeSpan ServerTimeoutInterval { get; set; } = TimeSpan.FromSeconds(15);
```

Comment:
Don't think we can change this name, it would be a breaking change and we generally don't do those except during major releases

Reply:
It did, look at the failing windows build

Reply:
Oh wait, we turned off appveyor 😆